### DISCUSSION TITLE
How to add Webcam?

### DISCUSSION THREAD

**Discussion by JasonNCH** [FYSETC/FYSETC-E4#22] [Q&A]
Hello Fysetc，

How to add webcam to this board? I see there is a wedcam module in the esp3d interface but not sure if it is supported by E4.

I also found the file transfer from ESP3D is rather slow, any suggestions to improve the speed?

**Comment by GerogeFu** [comment]
esp3d need to be installed in another esp32/esp8266 board, E4 use Marlin firmware with esp3Dlib but not esp3d firmware. And ESP3D is a bit slow at the moment because of the 2 pins UART speed, there is solution using 4 pins to communicate with motherboard should have faster speed, but it is still in development. Ask esp3d firmware writer Luc to have more details.
